### PR TITLE
Add user if logged in but does not exist

### LINKED ISSUE #76: Following mvn clean, user may be logged in but not exist

### PR DESCRIPTION
**Pull request by nbaldy**
Fixes #76 

Code summery:
Datastore - add getCurrentUser() function which checks if there is a user logged in, and if so creates that user if it does not exist in datastore and returns the current user. Returns null if no user is logged in. 
This function is more "robust" than userService.getCurrentUser() because it creates the user in datastore if the user does not yet exist. Therefore, all cases of "userService.getCurrentUser() or userService.isUserLoggedIn()" are replaced with this function. 
The "far-reach" of this is mostly removing the import of userService to non-datastore files (except login and logout servlets) and replacing appropriate methods with datastore.getCurrentUser()

### CODE REVIEW

**Comment by nbaldy** [comment]
Functionality can be tested by going to the admin page, and deleting the user which is currently logged in, here 
image
Then, user "is logged in but does not exist in datastore." 
If you visit the home page, user page, or country page the user should be created. It is also created on message post and aboutMe update.

File: src/main/java/com/google/codeu/servlets/UsersServlet.java [hunk 1]
```diff
@@ -27,36 +25,34 @@ public void init() {
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
 
-    UserService userService = UserServiceFactory.getUserService();
+    User currentUser = datastore.getCurrentUser();
     String requestUrl = request.getRequestURI();
-    String user = requestUrl.substring("/users/".length());
+    String userPageEmail = requestUrl.substring("/users/".length());
+    User loggedInUser = datastore.getCurrentUser();
+    User viewedUser = datastore.getUser(userPageEmail);
```
Comment (nbaldy) [inline src/main/java/com/google/codeu/servlets/UsersServlet.java:33]: Renames in userServlet -  
 - String **user** to **userPageEmail**
 - User **userData** to **viewedUser**

File: src/main/webapp/index.jsp [hunk 2]
```diff
@@ -20,9 +20,9 @@
 
     <%
       if (isUserLoggedIn) {
-        String username = (String) request.getAttribute("username");
+        String userEmail = (String) request.getAttribute("userEmail");
     %>
```
Comment (nbaldy) [inline src/main/webapp/index.jsp:24]: Rename in index.jsp:
 - String **username** to **userEmail**

**Review by drewbern — APPROVED** [review]
LGTM, as long as it has no conflicts with master!

**Pull request merged**